packages/cojson-transport-ws/src/index.ts

```diff
@@ -102,31 +81,37 @@ export function createWebSocketPeer({
         }
     });
 
+    async function pushMessage(msg: SyncMessage) {
+        if (websocket.readyState !== 1) {
+            await websocketOpen;
+        }
+
+        while (websocket.bufferedAmount > BUFFER_LIMIT && websocket.readyState === 1) {
+            await new Promise<void>((resolve) => setTimeout(resolve, BUFFER_LIMIT_POLLING_INTERVAL));
+        }
```

Comment:
For now it's ok to leave the buffering loop inside the push.

With the PeerState managing the backpressure we can be sure that there will be one loop active per peer.